### PR TITLE
Add business validation over project statuses

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related #416  , #381

### CODE REVIEW

File: src/resolvers/projectResolver.test.ts [hunk 1]
```diff
@@ -3276,6 +3356,86 @@ function projectBySlugTestCases() {
...
+        slug: project.slug,
+      },
+    });
+
+    assert.equal(
+      result.data.errors[0].message,
+      errorMessages.YOU_DONT_HAVE_ACCESS_TO_VIEW_THIS_PROJECT,
+    );
+  });
+  it('should not return cancelled project is user is loggedIn but is not owner of project', async () => {
```
Comment (aminlatifi) [inline src/resolvers/projectResolver.test.ts]: The description is repetitive, also `loggedIn` should be replaced by 'logged in'

File: src/resolvers/projectResolver.ts [hunk 2]
```diff
@@ -610,14 +604,7 @@ export class ProjectResolver {
...
 
-    if (
-      (project?.statusId === ProjStatus.drafted ||
-        project?.statusId === ProjStatus.cancelled) &&
-      // If project is draft or cancelled, just owner can view it
-      project?.admin !== String(user?.userId)
-    ) {
-      return null;
-    }
+    await canUserVisitProject(project, String(user?.userId));
```
Comment (aminlatifi) [inline src/resolvers/projectResolver.ts]: `canUserVisitProject` is not async
```suggestion
   canUserVisitProject(project, String(user?.userId));
```

**Review by aminlatifi — CHANGES_REQUESTED** [review]
Thanks @mohammadranjbarz 
Just a few syntax amendment

File: src/resolvers/projectResolver.test.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply src/resolvers/projectResolver.test.ts]: @aminlatifi  Thanks, done

File: src/resolvers/projectResolver.ts (hunk 2 above)
Comment (mohammadranjbarz) [reply src/resolvers/projectResolver.ts]: @amin Thanks done

**Pull request merged**